src/main/kotlin/org/malachite/estella/aplication/api/ApplicationController.kt

```diff
@@ -59,6 +65,32 @@ class ApplicationController(
     fun deleteApplication(@PathVariable applicationId: Int) =
         applicationService.deleteApplication(applicationId)
 
+    @CrossOrigin
+    @PutMapping("/{applicationId}/next")
+    fun updateApplicationStage(@RequestHeader(EStellaHeaders.jwtToken) jwt: String?, @PathVariable applicationId: Int): ResponseEntity<Message> {
+        applicationService.getApplicationById(applicationId).let {
+            recruitmentProcessService.getProcessFromStage(it.stage).let {
+                if (!checkUserRights(it.offer, jwt)) return OwnResponses.UNAUTH
+                return applicationService.setNextStageOfApplication(applicationId).let { OwnResponses.SUCCESS }
```

Comment:
letów nie trzeba zagnieżdżać. Możesz to zrobić tak, jakbyś w jsie robił theny na promisach, tzn
```kotlin
applicationService.getApplicationById(applicationId).let {
            recruitmentProcessService.getProcessFromStage(it.stage)
        }.let {
            if (!checkUserRights(it.offer, jwt)) return OwnResponses.UNAUTH
            return applicationService.setNextStageOfApplication(applicationId).let { OwnResponses.SUCCESS }
        }
```

Reply:
Tylko jest tu jeszcze jedna rzecz - nie powinno być to w kontrolerze. Kontrolery nie powinny mieć żadnej logiki, więc ten kod przeniósłbym do jakiegoś `ApplicationService#setNextStageInApplication(applicationId)`, czy coś takiego

Reply:
troche nie rozumiem, jedyna logika to sprawdzenie uprawnień i jest to analogiczne do tego co jest w innych kontrolerach, nawet po refaktorze